core/shared/src/main/scala/com/softwaremill/sttp/MonadError.scala

```diff
@@ -31,6 +31,14 @@ trait MonadAsyncError[R[_]] extends MonadError[R] {
   def async[T](register: (Either[Throwable, T] => Unit) => Unit): R[T]
 }
 
+object syntax {
+
+  implicit final class MonadErrorOps[R[_], A](val r: R[A]) extends AnyVal {
+    def map[B](f: A => B)(implicit ME: MonadError[R]): R[B] = ME.map(r)(f)
+    def flatMap[B](f: A => R[B])(implicit ME: MonadError[R]): R[B] = ME.flatMap(r)(f)
```

Comment:
are these used somewhere? as I wrote before, this might be a great addition, but maybe for a separate PR (if you could still bear with me ;) )

Reply:
I use it extensively in `AbstractFetchBackend`, it reads much nicer than `ME.flatMap`. I had a quick look previously and there wasn't any existing places where it could be used so it would be a PR with just these 4 lines, I don't know if it is worth it? I'll move it into a separate commit.

Reply:
Ok, let's leave it, I'll see if this can be used somewhere else as well later.